### DISCUSSION TITLE
[Feature]: Use locally storend images instead of fetching them from the server

### DISCUSSION THREAD

**Discussion by Peda1996** [immich-app/immich#1597] [Feature Request]
### Feature detail

Currently if an image is opened inside the App. It is fetched from the server. It doesn't matter if it is available locally or not.

It would be pretty neat, if the image is available locally, the local image is shown instead. This would save a lot of bandwith. 

### Platform

Mobile App

**Comment by SitramSoft** [comment]
It feels that everything is fetched from the server and there is nothing cached locally on the phone. It would be good to have a mechanism that caches thumbnails because at the moment navigating trough the mobile app with ~23k photos stored in various albums feels kinda sluggish.

Whenever the caching buffer gets filled, the items which have the oldest access timestamp, could be deleted to make room for new ones which are not yet in cache.

Some kind of service that runs in background could query the server periodically for a list of timestamps for the items in cache and if anything is changed server side, it will be deleted from cache. This way, next time the the item is accessed by the user, it will force a resync from the server.

**Comment by bo0tzz** [comment]
@alextran1502 we do this now, right?
